src/datasource.js

```diff
@@ -246,6 +277,35 @@ export class GenericDatasource {
     return graphs;
   }
 
+  /**
+   * Transform data returned by time series query into Grafana annotation format.
+   * @param results
+   * @param options
+   * @returns Array
+   */
+  static transformAnnotationResults(results, timeField, timeEndField, textField) {
+    const annotations = [];
+    results.forEach((result) => {
+      const responseObject = {};
+      responseObject.time = Number(result[timeField]) / 1000000;
+      if (!responseObject.time || responseObject.time.length === 0) {
```

Comment:
So if ``result[timeField]`` is not defined or similar, ``responseObject.time`` will contain NaN, right?

Also, I wonder why we check for ``time.length``. ``Number`` object doesn't contain ``length`` property, right?

Reply:
Yes it can contain `NaN` or `undefined`, this ends up working out with no errors thrown at all and simply not showing the annotation.

You're right about checking length on number making no sense, this was from an earlier iteration.